### DISCUSSION TITLE
Looking for rule on destructuring

### DISCUSSION THREAD

**Discussion by anzerr** [eslint/eslint#16268] [Q&A] [answered]
We've had a interesting bug in our code base. We where getting a "Maximum call stack size exceeded" when doing destructuring/spread on a function call. I looked over the rules but I didn't find any rules that would cover this case. Did I miss a rule that would help?

Example
``` javascript
class Report<E extends ReportLog> {

    private _logs: E[] = [];
    
    public addLogs(logs: E[]): Report<E> {
        this._logs.push(...logs)
        return this;
    }
    
}
```

short example (the value max is not the same for each env it's run in)
``` javascript
[].push(...Array(130000))
```

**Comment by mdjermanovic** [comment, marked as answer]
You can configure the [no-restricted-syntax](https://eslint.org/docs/latest/rules/no-restricted-syntax) rule to disallow spread syntax in arguments:

```
selector: "CallExpression > SpreadElement.arguments",
```

[Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlc3RyaWN0ZWQtc3ludGF4OiBbXCJlcnJvclwiLCB7XG4gICAgc2VsZWN0b3I6IFwiQ2FsbEV4cHJlc3Npb24gPiBTcHJlYWRFbGVtZW50LmFyZ3VtZW50c1wiLFxuICAgIG1lc3NhZ2U6IFwiRG9uJ3QgdXNlIGAuLi5gIGluIGZ1bmN0aW9uIGNhbGxzLlwiXG59XSovXG5cbmNsYXNzIFJlcG9ydCB7XG5cbiAgICBfbG9ncyA9IFtdO1xuICAgIFxuICAgIGFkZExvZ3MobG9ncykge1xuICAgICAgICB0aGlzLl9sb2dzLnB1c2goLi4ubG9ncylcbiAgICAgICAgcmV0dXJuIHRoaXM7XG4gICAgfVxuICAgIFxufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX0sInJ1bGVzIjp7fSwiZW52Ijp7ImVzNiI6dHJ1ZX19fQ==)
